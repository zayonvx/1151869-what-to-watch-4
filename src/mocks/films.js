export const moviesData = [
  {
    id: 1,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    image: `img/bohemian-rhapsody.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 3,
    title: `Macbeth`,
    image: `img/macbeth.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 4,
    title: `Aviator`,
    image: `img/aviator.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 5,
    title: `We need to talk about Kevin`,
    image: `img/we-need-to-talk-about-kevin.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 6,
    title: `What We Do in the Shadows`,
    image: `img/what-we-do-in-the-shadows.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 7,
    title: `Revenant`,
    image: `img/revenant.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
  {
    id: 8,
    title: `Johnny English`,
    image: `img/johnny-english.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterAlt: `The Grand Budapest Hotel poster`,
    rating: 8.9,
    ratingLevel: `Very good`,
    ratingCount: 240,
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe`,
  },
];
