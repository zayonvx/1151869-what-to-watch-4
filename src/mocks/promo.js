export const promoData = {
  title: `Shutter Island`,
  background: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  posterAlt: `The Grand Budapest Hotel poster`,
  genre: `Drama`,
  releaseDate: 2010,
};
