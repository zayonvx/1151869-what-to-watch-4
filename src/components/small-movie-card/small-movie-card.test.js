import React from "react";
import renderer from "react-test-renderer";
import SmallMovieCard from "./small-movie-card";
import {moviesData} from "../../mocks/films";

it(`SmallMovieCard is rendered`, () => {
  const tree = renderer
    .create(
        <SmallMovieCard
          movie={moviesData[0]}
          onTitleClick={() => {}}
          onHover={() => {}}
          index={0}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
