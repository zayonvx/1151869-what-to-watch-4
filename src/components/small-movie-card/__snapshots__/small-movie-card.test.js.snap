// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallMovieCard is rendered 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  data-id={1}
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height={175}
      muted={true}
      onCanPlayThrough={[Function]}
      poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      width={280}
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;
